startReactCommerce.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+git checkout main
+git pull
+git checkout -
+rm -rf node_modules
+nvm use
+npm install -g pnpm@8
+npm install -g yarn
+cp .env.local.example .env.local
+
+## Verdaccio setup
+read -p "Do you have a verdaccio account? (y/n)? [y]: " userInput
+
+# Remove leading/trailing whitespace and convert to lowercase
+userInput=$(echo "${userInput}" | tr '[:upper:]' '[:lower:]' | xargs)
+
+# Check the user's input
+while [[ ! "${userInput}" =~ ^(y|n)$ ]]; do
+    if [[ -z "${userInput}" ]]; then
+        userInput="y"  # Set default value to "yes" if no input is provided
+        break
+    fi
+    echo "Invalid input. Please enter 'y' or 'n'."
+    read -p "Do you have a verdaccio account? (y/n)? [y]: " userInput
+    userInput=$(echo "${userInput}" | tr '[:upper:]' '[:lower:]' | xargs)
+done
+
+# Continue based on the user's input
+if [[ "${userInput}" == "y" ]]; then
+    # Login to exisiting user:
+    pnpm login --registry=https://verdaccio.ubreakit.com    
+else
+    # Add new verdaccio user:
+    pnpm adduser --registry=https://verdaccio.ubreakit.com
+fi
+
+npmrcPath=~/.npmrc
+envFilePath=.env.local
+
+# Extract token from ~/.npmrc
+token=$(cat ~/.npmrc |  grep -o '\/\/verdaccio\.ubreakit\.com\/:_authToken="[^"]*"' | sed 's/\/\/verdaccio\.ubreakit\.com\/:_authToken="//' | sed 's/"$//')
+
+if [ -z "$token" ]; then
+    echo "Token not found in ~/.npmrc"
+    return 1
+fi
+
+# Update .env.local with VERDACCIO_AUTH_TOKEN
+awk -v token="\"$token\"" '/^VERDACCIO_AUTH_TOKEN=/{$0="VERDACCIO_AUTH_TOKEN=" token}1' .env.local > .env.local.tmp && mv .env.local.tmp .env.local
+
+echo -e "Token successfully copied to .env.local\n\n"
+
+cp backend/.env.example backend/.env
+
+envFilePath="frontend/.env.local"
+exampleEnvFilePath="frontend/.env.local.example"
+algoliaApiKey="\"12345-your-key-here\""
+
+# Check if .env.local file exists
+if [ ! -f "$envFilePath" ]; then
+    echo "$envFilePath does not exist. Creating a new file..."
+    cp "$exampleEnvFilePath" "$envFilePath"
+fi
+
+# Read ALGOLIA_API_KEY value from .env.local file
+apiKey=$(grep -o '^ALGOLIA_API_KEY=.*' "$envFilePath" | cut -d '=' -f 2)
+
+# Check if ALGOLIA_API_KEY value is empty
+if [ -z "$apiKey" ]; then
+    echo "ALGOLIA_API_KEY is empty. Filling in the value..."
+    awk -v placeholder="$algoliaApiKey" '/^ALGOLIA_API_KEY=/{gsub(/=.*/, "=" placeholder)} 1' "$envFilePath" > "$envFilePath.tmp" && mv "$envFilePath.tmp" "$envFilePath"
+fi
+
+pnpm install:all
+
+## We can remove the verdaccio token after exporting it to .env.local
+pnpm logout --registry=https://verdaccio.ubreakit.com
```

Comment:
Is there a reason to remove the token? Won't this cause problems for updating later?

Reply:
According to the readme:

⚠️ **_If you don't want to use the Verdaccio registry outside of this repo, you can run the following:_**

-  ```
   pnpm logout --registry=https://verdaccio.ubreakit.com
   ```
   -  This will remove the Verdaccio auth token from your `~/.npmrc`, and so long as you've exported the token to `.env.local`, you will still be able to install packages from Verdaccio for this repo.
   
   

We probably don't want to force the verdaccio config on everyone's computer so I think it's better to remove it. We will always export the verdaccio token on startup with this script so I don't think there's a reason so save it.

Reply:
Makes sense though I doubt people will use this script every time they want to run an update. Seems like it would take a very long time to delete node_modules and reinstall everything from scratch. But if it will work for a normal incremental install then it's probably fine.